core/src/main/scala/chisel3/Mem.scala

```diff
@@ -37,7 +37,7 @@ object Mem {
   ): Mem[T] = {
     requireIsChiselType(t, "memory type")
     val mt = t.cloneTypeFull
-    val mem = new Mem(mt, size)
+    val mem = new Mem(mt, size, sourceInfo)
```

Comment:
Try to keep PR's orthogonal. Getting `SourceInfo` properly piped through is good, but is a separate thing.

The way that I will usually do this is to:

1. Make this change.
2. Commit only this change.
3. Rebase the branch so that this commit is at the bottom.
4. Open a PR that includes just these changes.
5. Continue working and stack the PRs.

Reply:
I would prefer that this change remains in this PR. I need the sourceInfo to be passed in properly so that I can throw a `Builder.error` if someone tries to make a const mem. The error won't work without this.

Reply:
The point Schuyler is making is that you could open separate PRs for the notionally separate pieces even while still having the change available for this PR. It would be especially helpful if we wanted to backport that part of the change to previous branches while not backporting the Const work. I'm not going to push on it for this particular PR but it is a better development workflow that we all should strive to follow in general.